### DISCUSSION TITLE
help how to do custom weight for yolo11n

### DISCUSSION THREAD

**Discussion by SHtiyar** [ultralytics/ultralytics#21689] [Q&A]
i already train my model, because my dataset are inbalance so i use data augmentation to try to balanced the datasets, and after finised the training the performance for minor class is pretty bad under 50%, is there any possible way to do custom weighned class to improve the performance for minor class?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @SHtiyar, thanks for reaching out and for using Ultralytics 🚀! This is an automated response to help you get started. An Ultralytics engineer will also assist you here soon.

For general guidance, please check the Docs where you’ll find many helpful examples for training and usage:
- Python: https://docs.ultralytics.com/usage/python/
- CLI: https://docs.ultralytics.com/usage/cli/
- Main Docs: https://docs.ultralytics.com/

If this is a 🐛 Bug Report, please include a minimum reproducible example (MRE) to help us debug: https://docs.ultralytics.com/help/minimum-reproducible-example/

If this is a custom training ❓ Question (e.g., improving minority class performance in YOLO11n), please share:
- A brief description of your dataset and class distribution
- Representative images and labels
- Your training command/args, model, and relevant YAML files
- Training logs/plots and validation metrics per class
Also verify you’re following our Tips for Best Training Results: https://docs.ultralytics.com/guides/model-training-tips/

Join the Ultralytics community where it suits you best:
- Discord (real-time chat) 🎧: https://discord.com/invite/ultralytics
- Discourse (long-form Q&A): https://community.ultralytics.com/
- Subreddit: https://www.reddit.com/r/Ultralytics/

## Upgrade
Please upgrade to the latest ultralytics package with all requirements in a Python>=3.8 environment with PyTorch>=1.8 to ensure you benefit from the latest fixes and features:
```bash
pip install -U ultralytics
```
Requirements: https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml

## Environments
YOLO can be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):
- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See GCP Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/
- Amazon Deep Learning AMI. See AWS Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/
- Docker Image. See Docker Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/ <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all Ultralytics CI tests are currently passing. CI tests verify correct operation of all YOLO Modes and Tasks on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Thanks—per-class loss weighting isn’t exposed in YOLO11 today; to lift minority classes, oversample them (e.g., duplicate those images/paths in your train list), consider raising the global classification loss weight via the cls hyperparameter and training longer, and use close_mosaic to stabilize the last epochs; the Class Distribution guidance in Troubleshooting Common YOLO Issues and the Train settings page detail these options.

**Reply by SHtiyar** [reply]
i see, so what we can do is to balanced the datasets level and global classification loss, thank you.

**Reply by glenn-jocher** [reply]
Exactly; additionally ensure your validation split is representative (or use K-Fold Cross Validation with Ultralytics) and, if the minority objects are small, consider Image Tiling for Small Objects or a larger imgsz to improve recall.

**Reply by SHtiyar** [reply]
alraigh, now i use Leave-One-Out Cross Validation, and isolate the test data, i will try, thank you

**Reply by glenn-jocher** [reply]
Sounds good—LOOCV works well on small datasets; keep the held‑out test completely untouched for tuning, aggregate per‑class AP/mAP across folds using the model evaluation metrics, and if runtime becomes heavy consider stratified K‑Fold Cross Validation instead: model evaluation metrics and K‑Fold Cross Validation with Ultralytics.
